src/screens/TopicPageScreen/elements/ButtonFollowing.js

```diff
@@ -1,36 +1,42 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 import {View, TouchableNativeFeedback, Text, StyleSheet} from 'react-native';
 import {fonts, normalize, normalizeFontSize} from '../../../utils/fonts';
 import {COLORS} from '../../../utils/theme';
 
-const ButtonFollowing = ({handleSetUnFollow}) => {
+const ButtonFollowing = ({handleSetUnFollow, followType = 'signed'}) => {
   return (
     <TouchableNativeFeedback onPress={handleSetUnFollow}>
-      <View style={styles.buttonFollowing}>
-        <Text style={styles.textButtonFollowing}>Joined</Text>
+      <View style={styles.buttonFollowing(followType)}>
+        <Text style={styles.textButtonFollowing(followType)}>Joined</Text>
       </View>
     </TouchableNativeFeedback>
   );
 };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `ButtonFollowing` component now accepts a new prop `followType`. It's good that you've added PropTypes validation for this new prop. However, it would be better to define the possible values of `followType` using `PropTypes.oneOf()`, which can help prevent bugs due to typos or unexpected values.

```diff
ButtonFollowing.propTypes = {
  handleSetUnFollow: PropTypes.func,
- 19:   followType: PropTypes.string
+ 19:   followType: PropTypes.oneOf(['signed', 'incognito'])
};
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->